### ISSUE TITLE
extend_from_slice false positive

### ISSUE THREAD

**Issue by crumblingstatue** [Manishearth/rust-clippy#989]
Original code:

``` Rust
pub struct Foo;

pub fn foo(input: &[Foo]) -> Vec<&Foo> {
    let mut v = Vec::new();
    v.extend(input);
    v
}

```

Clippy's suggestion:

```
warning: use of `extend` to extend a Vec by a slice, #[warn(extend_from_slice)] on by default
 --> src/lib.rs:5:5
  |>
5 |>     v.extend(input);
  |>     ^^^^^^^^^^^^^^^
help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#extend_from_slice
help: try this
src/lib.rs:      v.extend_from_slice(input);

```

Compile error after applying suggestion:

```
error: the trait bound `Foo: std::clone::Clone` is not satisfied [--explain E0277]
 --> src/lib.rs:5:7
  |>
5 |>     v.extend_from_slice(input);
  |>       ^^^^^^^^^^^^^^^^^

error: mismatched types [--explain E0308]
 --> src/lib.rs:6:5
  |>
6 |>     v
  |>     ^ expected &-ptr, found struct `Foo`
note: expected type `std::vec::Vec<&Foo>`
note:    found type `std::vec::Vec<Foo>`

```

Note that there are two errors. One is `Clone` not being satisfied for `Foo`, the other is a type mismatch. `extend` produced `Vec<&Foo>`, extend_from_slice produces `Vec<Foo>`.

**Comment by llogiq** [comment]
We can now close this, as the lint in question is deprecated.
